### ISSUE TITLE
Modal Service: Add support for dynamic component

### ISSUE THREAD

**Issue by TheBr00m** [vikramlearning/blazorbootstrap#428]
Currently it seems the modal service does not support dynamic components as the modal component does. Dynamic components enable developers to add markup to the body.

For example, I would like to display key value pairs in a user friendly way.

N.B. I could not get the normal modal to work while nested deep in a component. Its logic seems to conflict with my Blazor Server application as the modal is visible but behind the transparent overlay, same as the rest of the page. So there is nothing left to interact with.

**Comment by FoobarLegend** [comment]
@gvreddy04 

I second this. The Modal component already provides the ShowAsync call with a generic type <T> (the childComponent to render). 
Adding a property for the ChildContent type in ModalOption (that gets passed to the ModalService) or making a generic ModalOption signature like ModalOption<ChildContent> would provide more flexibility of the content type to render, other than rendering only string based content. Also this means we can use the ModalService with one Modal component instance placed at the top most position in the DOM tree preventing a) unnecessary Modal instances within the DOM tree and thus better performance and b) nesting Modal instances within the DOM tree with dynamic rendered parent components can lead to CSS problems (mentioned by OP)

**Comment by talanw** [comment]
I agree this would be great. At the moment the fact that it's considered a string rather than something like a HTML string doesn't make it as useful as it could be.
